src/gui/DatabaseTabWidget.cpp

```diff
@@ -649,6 +649,46 @@ Database* DatabaseTabWidget::indexDatabase(int index)
     return nullptr;
 }
 
+QList<DatabaseWidget*> DatabaseTabWidget::getAllDatabaseWidgets(const bool onlyOpenDatabases) const
+{
+    QList<DatabaseWidget*> widgetList;
+    QHashIterator<Database*, DatabaseManagerStruct> i(m_dbList);
+    while (i.hasNext()) {
+        i.next();
+        if (onlyOpenDatabases && i.value().dbWidget->dbHasKey() && 
+            (i.value().dbWidget->currentMode() == DatabaseWidget::ViewMode || 
+             i.value().dbWidget->currentMode() == DatabaseWidget::EditMode)) {
+            widgetList.push_back(i.value().dbWidget);
+        } else if (!onlyOpenDatabases) {
+            widgetList.push_back(i.value().dbWidget);
+        }
+    }
+
+    return widgetList;
+}
+
+QString DatabaseTabWidget::getTabName(DatabaseWidget* dbWidget, const bool includeFilename) const
```

Comment:
It would be better to refactor so that the databaseWidget knows what it's filename is and you query it directly. We are perpetuating the terrible code that is databaseTabWidget...